### PR TITLE
Fixing flaky conversational test + flag it as a pipeline test.

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

@patrickvonplaten

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors which may be interested in your PR.

### CODE REVIEW

File: tests/test_pipelines_conversational.py [hunk 1]
```diff
@@ -52,9 +53,11 @@ def get_pipeline(self):
         # Force model output to be L
```
Comment (patrickvonplaten) [inline tests/test_pipelines_conversational.py:53]: what does "Force model output to be L" mean?

Comment (Narsil) [reply tests/test_pipelines_conversational.py:53]: The model is a random one that I generate on the fly.
Doing that enables me to actually make the output of that model consistent (because the logits are always 0, 0, 0, 1, 0, 0, ....)

I could always do what I did with the tokenizer which is move it to a dummy model on the hub and use AutoModel instead.
It does respect more previous testing

I used this here because makes that logic explicit in the test, so the tests gets easier to understand from a reader's perspective (is the "L" correct ? Can I change it etc..). I've seen quite a few tests in pipelines that were not detecting bugs because of hidden test logic.

I'll happily switch to AutoModel if you want to.
@LysandreJik

Comment (LysandreJik) [reply tests/test_pipelines_conversational.py:53]: I don't understand the L comment either

Comment (LysandreJik) [reply tests/test_pipelines_conversational.py:53]: Ah, it comes from the expected result:
```
                Conversation(
                    None, past_user_inputs=["What's the last book you have read?"], generated_responses=["L"]
                ),
```

**Review by LysandreJik — APPROVED** [review]
Works for me like that.

File: tests/test_pipelines_conversational.py (hunk 1 above)
Comment (Narsil) [reply tests/test_pipelines_conversational.py:53]: Yes
